app/models/generic_object_definition.rb

```diff
@@ -230,6 +360,69 @@ def check_not_in_use
   end
 
   def set_default_properties
-    self.properties = {:attributes => {}, :associations => {}, :methods => []} unless properties.present?
+    self.properties = {:attributes => {}, :attribute_constraints => {}, :associations => {}, :methods => []} unless properties.present?
+  end
+
+  private
+
+  def validate_default_value(attr_name, attr_type, value)
+    # Type check based on attribute type
+    case attr_type
+    when :boolean
+      unless [true, false].include?(value)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be a boolean")
+      end
+    when :integer
+      unless value.is_a?(Integer)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be an integer")
+      end
+    when :float
+      unless value.is_a?(Numeric)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be a number")
+      end
+    when :string
+      unless value.is_a?(String)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be a string")
+      end
+    when :datetime
+      unless value.is_a?(String) || value.is_a?(Time) || value.is_a?(DateTime)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be a valid datetime")
+      end
+    when :time
+      unless value.is_a?(String) || value.is_a?(Time)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be a valid time")
+      end
+    when :hash
+      unless value.is_a?(Hash)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be a Hash")
+      end
+    when :array
+      unless value.is_a?(Array)
+        errors.add(:properties, "default value for attribute [#{attr_name}] must be an Array")
+      end
+    end
+  end
+
+  def validate_enum_constraint(attr_name, attr_type, value)
+    unless value.is_a?(Array) && value.any?
```

Comment:
Minor but this is hard to read. Prefer:

```suggestion
    if value.is_a?(Array) && value.empty?
```